Type: StopTime is in VehicleJourney without indirection

TeXitoi opened the pull request:
This is the most instanciated object, thus we optimize it.

A Journey computation is 5% faster with this modification, and memory
almost the same (a little bit less).

l-vincent-l commented on source/ed/tests/adapted_test.cpp:
```diff
@@ -2242,10 +2242,10 @@ BOOST_AUTO_TEST_CASE(impact_stoppoint_4){
     AtAdaptedLoader loader;
     loader.apply(messages, *b.data->pt_data);
-    BOOST_CHECK_EQUAL(b.data->pt_data->stop_times.size(), 4);
+    //BOOST_CHECK_EQUAL(b.data->pt_data->stop_times.size(), 4);
```
can you make a stop_times_size() function?

TeXitoi committed:
ed: if the bounding shape crash, leave an error, but do not crash ed2nav

TeXitoi committed:
type: StopTime is in VehicleJourney without indirection

This is the most instanciated object, thus we optimize it.

A Journey computation is 5% faster with this modification, and memory
almost the same (a little bit less).

antoine-de commented on source/routing/dataraptor.h:
```diff
@@ -49,8 +49,8 @@ struct dataRAPTOR {
     std::vector<uint32_t> departure_times;
-    std::vector<type::StopTime*> st_idx_forward;// Nom a changer ce ne sont plus des idx mais des pointeurs
-    std::vector<type::StopTime*> st_idx_backward;
+    std::vector<const type::StopTime*> st_idx_forward;// Nom a changer ce ne sont plus des idx mais des pointeurs
```
I think it's time to change the comment :wink:

antoine-de replied:
Ok, it is not nice to ask you that, but I think some more comment about st_idx_forward and st_idx_backward would be nice 

(and maybe some explanation about why the pointer to a vector contained object is not that bad in this case)

antoine-de commented:
ok for me

TeXitoi committed:
type: externalize StopTime::comment

Small memory win (~5%), same performances

l-vincent-l commented on source/routing/dataraptor.h:
```diff
@@ -47,10 +47,24 @@ struct dataRAPTOR {
-    std::vector<type::StopTime*> st_idx_backward;
+
+    // arrival_times (resp. departure_times) are the different arrival
+    // (resp. departure) times of each stop points sorted by
```
stop time

TeXitoi committed:
routing: fix typo in comment
